### PR TITLE
Typing fixes for utilities

### PR DESCRIPTION
**Pull request by pomonam**
Add & fix typing for codes in `/algorithm_efficiency.` Also, minor fixes, such as removing old code and correcting comments. This PR should not change the behavior of the code at all.

### CODE REVIEW

File: algorithmic_efficiency/spec.py [hunk 1]
```diff
@@ -369,7 +351,33 @@ def init_optimizer_state(workload: Workload,
   pass
 
 
-_UpdateReturn = Tuple[OptimizerState, ParameterContainer, ModelAuxiliaryState]
+# def update_params(workload: Workload,
```
Comment (znado) [inline algorithmic_efficiency/spec.py:354]: can you delete this commented out code?

**Review by znado — APPROVED** [review]
looks great. just one small comment!

File: algorithmic_efficiency/spec.py (hunk 1 above)
Comment (pomonam) [reply algorithmic_efficiency/spec.py]: Thank you for the comment! I deleted the debugging comment.

**Comment by pomonam** [comment]
I will soon resolve the conflict.

**Comment by pomonam** [comment]
Done merging!

**Pull request merged**